torch/distributions/gumbel.py

```diff
@@ -31,14 +31,20 @@ def __init__(self, loc, scale, validate_args=None):
         self.loc, self.scale = broadcast_all(loc, scale)
         finfo = _finfo(self.loc)
         if isinstance(loc, Number) and isinstance(scale, Number):
-            batch_shape = torch.Size()
-            base_dist = Uniform(finfo.tiny, 1 - finfo.eps)
+            self._base_dist = Uniform(finfo.tiny, 1 - finfo.eps)
```

Comment:
Can you explain the pattern in this PR of using new attributes `._base_dist, ._transform` rather than the existing `.base_dist,.transform` attributes? I don't see why the existing attributes can be used instead.

Reply:
My bad; I missed that these were already available as these are subclassing `TransformedDistribution`. Will remove.